src/server/data-access/MwBotWikibaseMessagesRepo.ts

```diff
@@ -0,0 +1,86 @@
+import TechnicalProblem from '@/common/data-access/error/TechnicalProblem';
+import { MessageKeys } from '@/common/MessageKeys';
+import MessageTranslationCollection from '@/datamodel/MessageTranslationCollection';
+import MessageCollection from '@/datamodel/MessageCollection';
+import MessagesRepository from '@/common/data-access/MessagesRepository';
+import mwbot from 'mwbot';
+
+interface RespondMessage {
+	[ key: string]: string;
+}
+
+export default class MwBotWikibaseMessagesRepo implements MessagesRepository {
+	private bot: mwbot;
+	private messageKeys: string;
+
+	public constructor( bot: mwbot, messageKeys?: string[] ) {
+		this.bot = bot;
+		if ( messageKeys === null || typeof messageKeys === 'undefined' ) {
+			this.messageKeys = this.getRequestedKeys();
+		} else {
+			this.messageKeys = messageKeys.join( '|' );
+		}
+	}
+
+	public getMessagesInLanguage( inLanguage: string|null ): Promise<MessageTranslationCollection> {
+		if ( inLanguage == null ) {
+			return new Promise( ( resolve, reject ) => {
+				resolve( {} ) 
+			} );
+		}
+
+		return new Promise( ( resolve, reject ) => {
+			this.bot.request( {
+				action: 'query',
+				meta: 'allmessages',
+				ammessages: this.messageKeys,
+				amlang: inLanguage,
+			} )
+				.then( ( response: any ) => {
+					if ( !( 'query' in response ) ) {
+						reject( new TechnicalProblem( 'allmessages result not well formed.' ) );
+						return;
+					}
+
+					if ( !( 'allmessages' in response.query ) ) {
+						reject( new TechnicalProblem( 'allmessages result not well formed.' ) );
+						return;
+					}
+					resolve( this.getMessageTranslationCollection( inLanguage, response.query.allmessages ) );
+				} )
+				.catch( ( reason: string ) => {
+					reject( new TechnicalProblem( reason ) );
+				} );
+		} );
+	}
+
+	private getMessageTranslationCollection(
+		inLanguage: string,
+		response: RespondMessage[],
+	): MessageTranslationCollection {
+		const messageCollection: MessageTranslationCollection = {};
+		messageCollection[ inLanguage ] = this.transformMessages( response );
+		return messageCollection;
+	}
+
+	private transformMessages( response: RespondMessage[] ): MessageCollection {
+		const messageList: MessageCollection = {};
+		response.forEach( ( value: { [key: string]: string } ) => {
+			if ( value.hasOwnProperty( 'missing' ) ) {
+				messageList[ value.name ] = value.name; // this is just the same behavior like mw
```

Comment:
This situation can only occur if we have a message key configured that is not supported by mw. That is, this does not happen if a translation in a given language is missing in mw (in which case the English message is shown), but if the **key** is not present, right? So should this happen we are actually in a bad situation. I'm leaning towards actually throwing an exception.